src/Analyser/ExpressionTypeHolder.php

```diff
@@ -24,6 +24,11 @@ public static function createMaybe(Expr $expr, Type $type): self
 		return new self($expr, $type, TrinaryLogic::createMaybe());
 	}
 
+	public static function createNo(Expr $expr, Type $type): self
+	{
+		return new self($expr, $type, TrinaryLogic::createNo());
```

Comment:
This doesn't make sense. Instead of `no` just unset the entry in the array. This will also get rid of the error you commented about (the ErrorType problem).